src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx

```diff
@@ -16,15 +21,32 @@ interface DatepickerProps {
 export default function Datepicker({ page }: DatepickerProps) {
   const [state, dispatch] = useContext(AppContext);
   const pageState = state[page as keyof State] as PageState;
-  const [filterStartDate, filterEndDate] = pageState.filters.publish_date;
-  const [startDate, setStartDate] = useState(filterStartDate);
-  const [endDate, setEndDate] = useState(filterEndDate);
+  const [filterStartDate, filterEndDate]: Date[] = Array.from(pageState.filters.publish_date);
+  const [startDate, setStartDate] = useState<Date>(filterStartDate);
+  const [endDate, setEndDate] = useState<Date>(filterEndDate);
+  const minDate = state.calendarStartDates.minDate;
+  const maxDate = state.calendarStartDates.maxDate;
+  const [sliderMax, setSliderMax] = useState<number>((maxDate.getTime()-minDate.getTime())/86400000);
+  const [values, setValues] = useState<Date[]>([startDate, endDate]);
+
+  useEffect(() => {
+    setSliderMax((maxDate.getTime()-minDate.getTime())/86400000);
+  }, [maxDate, minDate])
 
   useEffect(() => {
     registerLocale("en", enUS)
     registerLocale("fr", fr)
   }, [])
 
+  useEffect(() => {
+    setValues([startDate, endDate]);
+  }, [startDate, endDate])
+
+  useEffect(() => {

```

Comment:
Is this useffect really necessary? If there is no way to access dates outside the min/max range in the UI, don't think guarding against this edge case is really a practical concern. 

Also if we were to guard against this edge case, we would probably want to ensure that out of range dates aren't even stored in the context (pageState.filters.publish_date) in the first place.

Reply:
Yeah you are right!

Reply:
Actually I noticed that although it isnt possible for the user to cause the above bug. The server loading speed does cause it initially which can cause the following 2 issues on first try:

![image](https://user-images.githubusercontent.com/29795835/128773826-5fce95e0-8eb9-4ecc-9738-8244abb0ec2f.png)

Where as you can see the dates on the slider are updated to the correct min and max, however the datepicker boxes have the old dates that are out of range. This causes another problem when you move the slider:

![image](https://user-images.githubusercontent.com/29795835/128773919-e5206036-69af-4a12-919f-b0507c33a3d2.png)

where it then sets the min date to the date that is out of range, and the blue bar runs off screen and detatches from the thumbs.

I changed the code a little so that the above check is only done when the server min and max are changed, but I dont think there is another way to get around this

Reply:
Fixed in b5cfb7a067d187dc9bcd1aef3fd4edfb98a416f0